ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/KnownPeers.hs

```diff
@@ -78,18 +78,14 @@ belowTarget actions
     -- We can only ask ones where we have not asked them within a certain time.
   , not (Set.null availableForPeerShare)
 
-    -- Are there any peers from the availableForPeerShare set we can really ask
-    -- (i.e. have correct PeerSharing permissions)?
-  , not (Set.null canAsk)
-
     -- No peer share requests should be issued when the node is in a sensitive
     -- state
   , not (requiresBootstrapPeers bootstrapPeersFlag ledgerStateJudgement)
   = Guarded Nothing $ do
       -- Max selected should be <= numPeerShareReqsPossible
       selectedForPeerShare <- pickPeers st
                               policyPickKnownPeersForPeerShare
-                              canAsk
+                              availableForPeerShare
```

Comment:
This might pick peers which don't have peer sharing enabled resulting in a protocol failure

Reply:
No, see previous comment regarding `availableForPeerShare`.